src/main/cpp/mechanisms/DragonTale/GrabAlgaeFloorState.cpp

```diff
@@ -76,6 +76,6 @@ bool GrabAlgaeFloorState::AtTarget()
 bool GrabAlgaeFloorState::IsTransitionCondition ( bool considerGamepadTransitions )
 {
 	// To get the current state use m_mechanism->GetCurrentState()
-	return false;
-	// return (considerGamepadTransitions && TeleopControl::GetInstance()->IsButtonPressed(TeleopControlFunctions::EXAMPLE_MECH_FORWARD));
+
+	return (considerGamepadTransitions && (TeleopControl::GetInstance()->IsButtonPressed(TeleopControlFunctions::ALGAE_INTAKE)) && m_mechanism->IsAlgaeMode() && (m_mechanism->GetAlgaeSensorState()));
```

Comment:
You need to do this if the AlgaeSensorState is false (add a Not (!) in front of the m_mechanism->GetAlgaeSensorState())